docker/bookkeeper/entrypoint.sh

```diff
@@ -8,68 +8,45 @@
 #
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
+set -e
 
-set -eo pipefail
-
-ZK_HOME=/opt/zookeeper
-BK_HOME=/opt/bookkeeper
-
-PORT0=${PORT0:-$bookiePort}
-PORT0=${PORT0:-3181}
-ZK_URL=${ZK_URL:-127.0.0.1:2181}
-ZK_URL=$(echo "${ZK_URL}" | sed -r 's/,/;/g')
-USE_MOUNT=${USE_MOUNT:-0}
+BOOKIE_PORT=${bookiePort:-${BOOKIE_PORT}}
+BOOKIE_PORT=${BOOKIE_PORT:-3181}
+BK_zkServers=$(echo "${ZK_URL:-127.0.0.1:2181}" | sed -r 's/;/,/g')
+ZK_URL=$(echo "${ZK_URL:-127.0.0.1:2181}" | sed -r 's/,/;/g')
 PRAVEGA_PATH=${PRAVEGA_PATH:-"pravega"}
 PRAVEGA_CLUSTER_NAME=${PRAVEGA_CLUSTER_NAME:-"pravega-cluster"}
 BK_CLUSTER_NAME=${BK_CLUSTER_NAME:-"bookkeeper"}
-BK_AUTORECOVERY=${BK_AUTORECOVERY:-"false"}
-BK_useHostNameAsBookieID=${BK_useHostNameAsBookieID:-"false"}
-
 BK_LEDGERS_PATH="/${PRAVEGA_PATH}/${PRAVEGA_CLUSTER_NAME}/${BK_CLUSTER_NAME}/ledgers"
-
-if [ $USE_MOUNT -eq 0 ]; then
-    BK_DIR="/bk"
-else
-    BK_DIR=$MESOS_SANDBOX
-fi
-
-echo "bookie service port0 is $PORT0 "
-echo "ZK_URL is $ZK_URL"
-echo "BK_DIR is $BK_DIR"
-echo "BK_LEDGERS_PATH is $BK_LEDGERS_PATH"
-
-sed -i "s|bookiePort=.*\$|bookiePort=${PORT0}|" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|metadataServiceUri=.*\$|metadataServiceUri=zk://${ZK_URL}${BK_LEDGERS_PATH}|" ${BK_HOME}/conf/bk_server.conf
-sed -i "/zkServers=.*/d" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|journalDirectory=.*\$|journalDirectory=${BK_DIR}/journal|" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|ledgerDirectories=.*\$|ledgerDirectories=${BK_DIR}/ledgers|" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|indexDirectories=.*\$|indexDirectories=${BK_DIR}/index|" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|# autoRecoveryDaemonEnabled=.*\$|autoRecoveryDaemonEnabled=${BK_AUTORECOVERY}|" ${BK_HOME}/conf/bk_server.conf
-sed -i "s|# useHostNameAsBookieID=.*\$|useHostNameAsBookieID=${BK_useHostNameAsBookieID}|" ${BK_HOME}/conf/bk_server.conf
-
-echo "
-tlsProvider=OpenSSL
-# key store
-tlsKeyStoreType=JKS
-tlsKeyStore=/var/private/tls/bookie.keystore.jks
-tlsKeyStorePasswordPath=/var/private/tls/bookie.keystore.passwd
-# trust store
-tlsTrustStoreType=JKS
-tlsTrustStore=/var/private/tls/bookie.truststore.jks
-tlsTrustStorePasswordPath=/var/private/tls/bookie.truststore.passwd" >> ${BK_HOME}/conf/bk_server.conf
+BK_DIR="/bk"
+
+export BOOKIE_PORT=${BOOKIE_PORT}
+export BK_zkServers=${BK_zkServers}
+export BK_metadataServiceUri=zk://${ZK_URL}${BK_LEDGERS_PATH}
+export BK_journalDirectory=${BK_DIR}/journal
+export BK_ledgerDirectories=${BK_DIR}/ledgers
+export BK_indexDirectories=${BK_DIR}/index
+export BK_CLUSTER_ROOT_PATH=/${PRAVEGA_PATH}/${PRAVEGA_CLUSTER_NAME}/${BK_CLUSTER_NAME}
+
+export BK_tlsProvider=OpenSSL
```

Comment:
Why are we setting these security options like this? As far as I can tell, we are hardwiring the security configuration here, which is possibly not a good idea. cc/ @ravisharda

Reply:
In the code below, keystores, truststores and their passwords are being dealt with (except for in line 35). 

https://github.com/pravega/pravega/blob/5af078f0f353fe7b74ba63f5d1336ff55ec6167f/docker/bookkeeper/entrypoint.sh#L33-L37

Questions:

1. I haven't got a chance to spend  enough time looking at it, but does it also imply that these files are baked in the image? Does that mean that if these need to be replaced by user-supplied material, the image would need to change? 

2. Should we consider other options for passing these sensitive material? Some of the popular options these days are: 
   * As secrets in files via a volume mounted on the container, 
   * Using secrets management facilities like [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) or  [Docker Swarm](https://docs.docker.com/engine/swarm/secrets/)

Reply:
Those TLS settings were hardwired in https://github.com/pravega/pravega/pull/2302 in an attempt to secure segmentstore-bookie communication, the only thing that changed in this PR is how it's done: before, they were appended to `bk_server.conf`; now, via env vars. The result should be the same.

I also think that it's not a good idea to hardwire them, however, I'm not sure if we should just use that configuration as the default, so that users can override it, or do something else. What do you think @fpj @ravisharda ?

Reply:
Issue https://github.com/pravega/pravega/issues/3390 has been create to discuss this.